### ISSUE TITLE
onSnapToItem callback not triggering consistently

### ISSUE THREAD

**Issue by odusseys** [archriss/react-native-snap-carousel#153]
Version 3.2.2
I am trying to use the onSnapToItem callback to track the index of the carousel. My code is : 

```
<Carousel
          style={{ flex: 1 }}
          ref={c => {
            if (c) this._carousel = c;
          }}
          data={this._cardData}
          renderItem={this._makeCard}
          sliderWidth={sliderWidth}
          itemWidth={itemWidth}
          inactiveSlideOpacity={0.0}
          onSnapToItem={index => {
            console.warn(index);
            this.setState({ index });
          }}
        />
```

The callback is not consistently called. It is consistently called when calling this._carousel.snapToNext/Prev, but not always when swiping (in particular almost never when swiping left).

**Comment by bd-arc** [comment]
Hi @odusseys,

Have you read [this note](https://github.com/archriss/react-native-snap-carousel/#unreliable-callbacks) from the doc? Also, are you currently [testing on Android with dev mode enabled](https://github.com/archriss/react-native-snap-carousel/#android-performance)?

**Comment by odusseys** [comment]
Your first link is not working for me, it's just redirecting me to the
repo. Also I am testing on the iOS simulator.

**Comment by odusseys** [comment]
Just saw that, thanks. I tried 
- enabling JS debugging : no change
- setting a higher value for callbackOffsetMargin (tried 10, 50, 500), no change

**Comment by bd-arc** [comment]
Hum, that's odd. I'm using the latest version in a store app and it works flawlessly.

Can you try:
- Running [the example](https://github.com/archriss/react-native-snap-carousel/tree/master/example)? If the pagination works, then the callback is also working.
- Deploy a production version of your app on a real device with an `alert()` callback for example?

**Comment by bd-arc** [comment]
@odusseys I think I've been able to reproduce the issue. Can you confirm that it happens only when you do a long swipe instead of a short swipe?

**Comment by bd-arc** [comment]
@odusseys Indeed, I made a mistake while refactoring the callback mechanism. [This commit](https://github.com/archriss/react-native-snap-carousel/commit/e075e124124d70018853fde9346828ed37138495) fixes the issue.

Can you confirm that it works for you?

**Comment by wildvalcom** [comment]
@bd-arc I've got the same problem. It seems that this only happens with long swipe.

**Comment by bd-arc** [comment]
@wildvalcom Have you tried the commit referenced in my latest comment?

**Comment by odusseys** [comment]
Has this been released to npm or should I link the repo directly ?

**Comment by peterchibunna** [comment]
I had the issue describe by OP. It seems to go away when you add this config properties
```
enableMomentum={false}
enableSnap={true}
decelerationRate={'fast'}
```

**Comment by bd-arc** [comment]
As previously stated, I've already pushed [a commit](https://github.com/archriss/react-native-snap-carousel/commit/edd98f36cf68e6a953881244294eaeaa69bde255) for that. You can try it this way:
```
"react-native-snap-carousel": "https://github.com/archriss/react-native-snap-carousel#edd98f3"
```

@odusseys @wildvalcom @peterchibunna Does it solve the issue for you?

**Comment by wildvalcom** [comment]
@bd-arc thanks mate, works good for me.

**Comment by bd-arc** [comment]
@wildvalcom Perfect! Thanks for letting me know.

The fix is now available in version `3.2.3`.
